### PR TITLE
[OPPRO-267] Add sort operator

### PR DESCRIPTION
**Pull request by JkSelf**
## What changes were proposed in this pull request?
Add sort operator

## How was this patch tested?

Existing tests in [TestOperator](https://github.com/oap-project/gluten/blob/c3d4f9d56f36df381bf080e19307624b1db8b061/backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala#L168)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: backends-velox/src/main/scala/io/glutenproject/execution/VeloxNativeColumnarToRowExec.scala [hunk 1]
```diff
@@ -101,8 +99,12 @@ case class VeloxNativeColumnarToRowExec(child: SparkPlan)
         } else {
           var info: NativeColumnarToRowInfo = null
           val beforeConvert = System.nanoTime()
+          ConverterUtils.printBatch(ArrowColumnarBatches.ensureLoaded(
+            SparkMemoryUtils.contextArrowAllocator(), batch))
```
Comment (rui-mo) [inline backends-velox/src/main/scala/io/glutenproject/execution/VeloxNativeColumnarToRowExec.scala:103]: Should this be removed?

**Comment by rui-mo** [comment]
Generally LGTM. I remember there are some cases for sort keys being expression. Does this PR support that? If not, we may mark the gap in TODO. For example, `sort(cast(c0), c1/c2)`.

**Comment by JkSelf** [comment]
> Generally LGTM. I remember there are some cases for sort keys being expression. Does this PR support that? If not, we may mark the gap in TODO. For example, `sort(cast(c0), c1/c2)`.

This PR support the sort key being expression case. Added new ut with expression sort key.

**Comment by rui-mo** [comment]
Does a Project Rel being inserted before sort for the expressions?

**Comment by JkSelf** [comment]
> Does a Project Rel being inserted before sort for the expressions?

Yes. Added a Project Rel when the sort key is expression. Can you help to review again?

**Comment by rui-mo** [comment]
Does current implementation for preProject change the data to sort? I remember the original data should not be changed. For example, sort(upper(col_0)) will use the uppered data to sort, but the sorted data is still the original col_0.

**Comment by JkSelf** [comment]
> Does current implementation for preProject change the data to sort? I remember the original data should not be changed. For example, sort(upper(col_0)) will use the uppered data to sort, but the sorted data is still the original col_0.

Added post project to removing the sorting columns and only remain the original columns.

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**